client/src/components/users/UserProfile.tsx

```diff
@@ -0,0 +1,184 @@
+    constructor(props: Props) {
+        super(props);
+        console.log(this.props.preloadUser);
+        this.state = {
+            user: this.props.preloadUser,
+        };
+    }
+
+    public handleInputChange = (event: ChangeEvent<HTMLInputElement>): void => {
+        const target = event.target;
+        let value: string | number = target.value;
+        const name = target.name;
+        const inputType = target.type;
+        // Convert number input values to numbers
+        if (inputType === "number") {
+            value = Number.parseFloat(value);
+        }
+
+        // @ts-ignore
+        this.setState({
+            user: {
+                ...this.state.user,
+                [name]: value
+            }
+        });
+    }
+
+    public handleInputChangeCheckbox = (event: FormEvent<HTMLInputElement>, checkboxProps: CheckboxProps): void => {
+        if (checkboxProps && checkboxProps.name && typeof checkboxProps.checked !== "undefined") {
+            const value: boolean = checkboxProps.checked;
+            const name: string = checkboxProps.name;
+
+            // @ts-ignore
+            this.setState({
+                user: {
+                    ...this.state.user,
+                    [name]: value
+                }
+            });
+        }
+    };
+
+    public render() {
+        const isAdmin: boolean = !!this.props.user && this.props.user.admin;
+        const adminHeader = isAdmin ? <Header size="medium">User Administration</Header> : "";
+        const haveId = isAdmin ? <Popup inverted={true} trigger={<Form.Checkbox name="haveID"
+                                                                                label="Have ID"
+                                                                                checked={this.state.user.haveID}
+                                                                                onChange={this.handleInputChangeCheckbox}
+            />}
+                                        content="Whether the hardware desk currently has possession of this user's photo ID as collateral"/>
+            : "";
+        const userIsAdmin = isAdmin ? <Popup inverted={true} trigger={<Form.Checkbox label="Admin"
+                                                                                     name="admin"
+                                                                                     checked={this.state.user.admin}
+                                                                                     onChange={this.handleInputChangeCheckbox}
+            />}
+                                             content="Whether this user can access administration pages like this one"/>
+            : "";
+
+        const adminUuid = isAdmin ?
+            <Form.Field width={6}>
+                <label>UUID</label>
+                <p>{this.state.user.uuid}</p>
+            </Form.Field>
+            : "";
+
+        let yourThis = "your";
```

Comment:
these string optimizations  are a little :sweat_smile: err

Reply:
I'll chill a bit in the future

Reply:
Thought more about this.  Good points.  I will change so the success message is more concise

Reply:
Was able to modify the language to make it less specific without losing anything